src/govuk/components/character-count/character-count.mjs

```diff
@@ -23,14 +24,49 @@ import { mergeConfigs, normaliseDataset } from '../../common.mjs'
  * @param {Number} [config.threshold=0] - The percentage value of the limit at
  * which point the count message is displayed. If this attribute is set, the
  * count message will be hidden by default.
+ * @param {Object} [config.i18n]
+ * @param {String} [config.i18n.charactersUnderLimitOne="You have %{count} character remaining"]
+ *   Message notifying users they're 1 character under the limit
+ * @param {String} [config.i18n.charactersUnderLimitOther="You have %{count} characters remaining"]
+ *   Message notifying users they're any number of characters under the limit
+ * @param {String} [config.i18n.charactersAtLimit="You have no characters remaining"]
+ *   Message notifying users they've reached the limit number of characters
+ * @param {String} [config.i18n.charactersOverLimitOne="You have %{count} character too many"]
+ *   Message notifying users they're 1 character over the limit
+ * @param {String} [config.i18n.charactersOverLimitOther="You have %{count} characters too many"]
+ *   Message notifying users they're any number of characters over the limit
+ * @param {String} [config.i18n.wordsUnderLimitOne="You have %{count} word remaining"]
+ *   Message notifying users they're 1 word under the limit
+ * @param {String} [config.i18n.wordsUnderLimitOther="You have %{count} words remaining"]
+ *   Message notifying users they're any number of words under the limit
+ * @param {String} [config.i18n.wordsAtLimit="You have no words remaining"]
+ *   Message notifying users they've reached the limit number of words
+ * @param {String} [config.i18n.wordsOverLimitOne="You have %{count} word too many"]
+ *   Message notifying users they're 1 word over the limit
+ * @param {String} [config.i18n.wordsOverLimitOther="You have %{count} words too many"]
+ *   Message notifying users they're any number of words over the limit
  */
 function CharacterCount ($module, config) {
   if (!$module) {
     return this
   }
 
   var defaultConfig = {
-    threshold: 0
+    threshold: 0,
+    i18n: {
+      // Characters
+      charactersUnderLimitOne: 'You have %{count} character remaining',
+      charactersUnderLimitOther: 'You have %{count} characters remaining',
+      charactersAtLimit: 'You have no characters remaining',
```

Comment:
Just a note that the two 'AtLimit' strings here are different to what these components currently return ("You have 0 characters remaining" and "You have 0 words remaining"). Not sure if that needs to be flagged as a breaking change?

Reply:
I was discussing the spelling with Claire, and I think I'll revert to the 0 and find a different way to test.

From what I understood of what makes a breaking change or not (which I may well still be blurry about), this wouldn't qualify as users wouldn't have to change their code when upgrading, it's "only" the HTML we output that would change.

Reply:
That makes sense to me. 

Does make me think more generally though: in future, if we change what a string says in English, we may want to flag it in the release notes in case it warrants a change to translations too.

Reply:
That's a very good point for future changelogs indeed.